manager/integration/tests/backupstore.py

```diff
@@ -0,0 +1,410 @@
+def nfs_write_backup_cfg_file():
+    raise NotImplementedError
+
+
+def minio_write_backup_cfg_file(client, core_api, volume_name, backup_name, backup_cfg_data): # NOQA
+    secret_name = backupstore_get_secret(client)
+    assert secret_name != ''
+
+    minio_api = minio_get_api_client(client, core_api, secret_name)
+    bucket_name = minio_get_backupstore_bucket_name(client)
+    minio_backup_cfg_file_path = minio_get_backup_cfg_file_path(volume_name,
+                                                                backup_name)
+
+    tmp_backup_cfg_file = "/tmp/backup_" + backup_name + ".cfg"
+    with open(tmp_backup_cfg_file, 'w') as tmp_bkp_cfg_file:
+        tmp_bkp_cfg_file.write(str(backup_cfg_data))
+
+    try:
+        with open(tmp_backup_cfg_file, 'rb') as tmp_bkp_cfg_file:
+            tmp_bkp_cfg_file_stat = os.stat(tmp_backup_cfg_file)
+            minio_api.put_object(bucket_name,
+                                 minio_backup_cfg_file_path,
+                                 tmp_bkp_cfg_file,
+                                 tmp_bkp_cfg_file_stat.st_size)
+    except ResponseError as err:
+        print(err)
+
+
+def backupstore_delete_backup_cfg_file(client, core_api, volume_name, backup_name):  # NOQA
+    backupstore = backupstore_get_backup_target(client)
+
+    if is_backupTarget_s3(backupstore):
+        return minio_delete_backup_cfg_file(client,
+                                            core_api,
+                                            volume_name,
+                                            backup_name)
+
+    elif is_backupTarget_nfs(backupstore):
+        return nfs_delete_backup_cfg_file()
+
+
+# TODO: implement nfs_delete_backup_cfg_file
+def nfs_delete_backup_cfg_file():
+    raise NotImplementedError
+
+
+def minio_delete_backup_cfg_file(client, core_api, volume_name, backup_name):
+    secret_name = backupstore_get_secret(client)
+    assert secret_name != ''
+
+    minio_api = minio_get_api_client(client, core_api, secret_name)
+    bucket_name = minio_get_backupstore_bucket_name(client)
+    minio_backup_cfg_file_path = minio_get_backup_cfg_file_path(volume_name,
+                                                                backup_name)
+
+    try:
+        minio_api.remove_object(bucket_name, minio_backup_cfg_file_path)
+    except ResponseError as err:
+        print(err)
+
+
+def backupstore_delete_volume_cfg_file(client, core_api, volume_name):  # NOQA
+    backupstore = backupstore_get_backup_target(client)
+
+    if is_backupTarget_s3(backupstore):
+        return minio_delete_volume_cfg_file(client,
+                                            core_api,
+                                            volume_name)
+
+    elif is_backupTarget_nfs(backupstore):
+        return nfs_delete_volume_cfg_file()
+
+
+# TODO: implement nfs_delete_volume_cfg_file
+def nfs_delete_volume_cfg_file():
+    raise NotImplementedError
+
+
+def minio_delete_volume_cfg_file(client, core_api, volume_name):
```

Comment:
The code for volume and backup cfg deletion currently is duplicated.
You could have a single minio_delete_file(client, core_api, path) call.
That takes care of all the secret stuff.
Then the delete_volume_cfg_file and delete_backup_cfg_file code only passes the path of the relevant .cfg

Reply:
fixed